exchanges/stream/websocket.go

```diff
@@ -826,12 +1072,19 @@ func (w *Websocket) SubscribeToChannels(subs subscription.List) error {
 
 // AddSubscriptions adds subscriptions to the subscription store
 // Sets state to Subscribing unless the state is already set
-func (w *Websocket) AddSubscriptions(subs ...*subscription.Subscription) error {
+func (w *Websocket) AddSubscriptions(conn Connection, subs ...*subscription.Subscription) error {
 	if w == nil {
 		return fmt.Errorf("%w: AddSubscriptions called on nil Websocket", common.ErrNilPointer)
 	}
-	if w.subscriptions == nil {
-		w.subscriptions = subscription.NewStore()
+	var subscriptionStore **subscription.Store
+	if candidate, ok := w.connections[conn]; ok {
+		subscriptionStore = &candidate.Subscriptions
+	} else {
+		subscriptionStore = &w.subscriptions
+	}
+
+	if *subscriptionStore == nil {
+		*subscriptionStore = subscription.NewStore()
```

Comment:
Could you explain the need for the pointer to a pointer situation?

Reply:
Just keeps the code more concise, there is nothing that it adds just removes a bunch of doubled up code when it is nil and needs a new store.

Reply:
Ahh okay the core issue is when `subscriptionStore` is nil. We then call:
```go
subscriptionStore = subscription.NewStore()
```

But that still needs to be set somewhere and when this function returns it just disappears. So what the ** means is that we are setting up a pointer to a field value.